Frontend/ui-customer/src/Helpers/Utils.purs

```diff
@@ -327,4 +334,22 @@ getCurrentLocationMarker :: String -> String
 getCurrentLocationMarker currentVersion = if isPreviousVersion currentVersion (getPreviousVersion "") then "ic_customer_current_location" else "ny_ic_customer_current_location"
 
 getPreviousVersion :: String -> String 
-getPreviousVersion _ = if os == "IOS" then "1.2.5" else "1.2.0"
\ No newline at end of file
+getPreviousVersion _ = if os == "IOS" then "1.2.5" else "1.2.0"
+
+foreign import getMerchantId :: String -> Foreign
+
+data Merchant = NAMMAYATRI | JATRISAATHI | YATRI
+
+derive instance genericMerchant :: Generic Merchant _
+instance eqMerchant :: Eq Merchant where eq = genericEq
+instance showMerchant :: Show Merchant where show = genericShow
+instance encodeMerchant :: Encode Merchant where encode = defaultEnumEncode
+instance decodeMerchant:: Decode Merchant where decode = defaultEnumDecode
+
+getMerchant :: String -> Merchant
+getMerchant _ = case (decodeMerchantId (getMerchantId "")) of 
```

Comment:
dont use `_` and you can use LazyCheck constructor instead of passing empty string anywhere